rust/relay/src/udp_socket.rs

```diff
@@ -0,0 +1,54 @@
+use anyhow::Result;
+use std::net::SocketAddr;
+use std::task::{ready, Context, Poll};
+use tokio::io::ReadBuf;
+
+const MAX_UDP_SIZE: usize = 65536;
+
+/// A thin wrapper around [`tokio::net::UdpSocket`] that provides a slightly more convenient API.
```

Comment:
I feel like it'd be less code for us to pass the buffer to the `EventLoop` rather than having this wrapper.

Reply:
I went back and forth about 3 times but ended up settling on this because it is more convenient to use and I need it in two places (allocation and main socket). Feel free to experiment yourself!